api/v1beta1/cloudstackaffinitygroup_types.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	AffinityGroupFinalizer = "cloudstackcluster.infrastructure.cluster.x-k8s.io"
+)
+
+// CloudStackAffinityGroupSpec defines the desired state of CloudStackAffinityGroup
+type CloudStackAffinityGroupSpec struct {
+	// Mutually exclusive parameter with AffinityGroupIDs.
+	// Can be "host affinity" or "host anti-affinity". Will create an affinity group per machine set.
+	Type string `json:"type,omitempty"`
+
+	// Name.
+	Name string `json:"name,omitempty"`
+
+	// ID.
+	//+optional
+	ID string `json:"id,omitempty"`
```

Comment:
Indeed we do.

I'll transition us to use this. One thing to note is that initializing struct might have looked like:
``` Go
affGroup := &CloudStackAffinityGroup{Name: SomeName, ID: SomeID}
```
But with the identity ref it has to look like:

``` Go
affGroup := &CloudStackAffinityGroup{
    CloudStackResourceIdentifier:{
            Name: SomeName, 
            ID: SomeID}}
```

But I still think it's worth it.

Reply:
I made changes to things that did make sense, but it doesn't make sense to use this everywhere. If the identifier type is used, the json needs a name for the node.

In this case it wouldn't make sense because we'd need something like `affinityGroup.ident.name`